src/main/java/org/checkstyle/autofix/CheckstyleRecipeRegistry.java

```diff
@@ -47,28 +56,49 @@ private CheckstyleRecipeRegistry() {
      * using the simple name of the check, and applies the factory to generate Recipe instances.
      *
      * @param violations the list of Checkstyle violations
+     * @param config the checkstyle configuration
      * @return a list of generated Recipe objects
      */
-    public static List<Recipe> getRecipes(List<CheckstyleViolation> violations) {
+    public static List<Recipe> getRecipes(List<CheckstyleViolation> violations,
+                                          CheckConfiguration config) {
+        return violations.stream()
+                .collect(Collectors.groupingBy(CheckstyleViolation::getSource))
+                .entrySet()
+                .stream()
+                .map(entry -> createRecipe(entry, config))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private static Recipe createRecipe(Map.Entry<String, List<CheckstyleViolation>> entry,
+                                       CheckConfiguration config) {
 
-        final Map<String, List<CheckstyleViolation>> violationsByCheck = violations.stream()
-                .collect(Collectors.groupingBy(CheckstyleViolation::getSource));
+        final String simpleCheckName = normalizeCheckName(entry.getKey());
+        final List<CheckstyleViolation> violations = entry.getValue();
 
-        final List<Recipe> recipes = new ArrayList<>();
+        return Optional.ofNullable(RECIPE_MAP_WITH_CONFIG.get(simpleCheckName))
+                .map(factory -> {
+                    return factory.apply(violations,
+                            extractCheckConfiguration(config, simpleCheckName));
+                }).orElseGet(() -> {
+                    return Optional.ofNullable(RECIPE_MAP.get(simpleCheckName))
+                            .map(factory -> factory.apply(violations))
+                            .orElse(null);
+                });
```

Comment:
could you use just simple if-else to avoid this ofNullable and orElse complexity?